pandas/io/gbq.py

```diff
@@ -160,7 +160,48 @@ def get_credentials(self):
         if self.private_key:
             return self.get_service_account_credentials()
         else:
-            return self.get_user_account_credentials()
+            # Try to retrieve Application Default Credentials
+            credentials = self.get_application_default_credentials()
+            if not credentials:
+                credentials = self.get_user_account_credentials()
+            return credentials
+
+    def get_application_default_credentials(self):
+        """
+        This method tries to retrieve the "default application credentials"
+        Could be useful for running code on Google Cloud Platform
+        """
+        oauth2client_library_imported = False
+        try:
```

Comment:
are these libararies a newer / different version that we are currently importing?